web/src/components/import/custom/header.js

```diff
@@ -1,31 +1,41 @@
 import React from 'react';
+import {styled} from '@mui/material/styles';
 import PropTypes from 'prop-types';
-import {makeStyles} from '@mui/styles';
 
-const useStyles = makeStyles((theme) => ({
-  fishSize: {
+const PREFIX = 'AgGridHeader';
+
+const classes = {
+  fishSize: `${PREFIX}-fishSize`,
+  invertSize: `${PREFIX}-invertSize`
+};
```

Comment:
I still think you need to work out a way to clean this,

```javascript
const classes = {
  fishSize: `${PREFIX}-fishSize`,
  invertSize: `${PREFIX}-invertSize`
};
```

the same thing appears in `GridHeader.js`

```javascript
const classes = {
  fishSize: `${PREFIX}-fishSize`,
  invertSize: `${PREFIX}-invertSize`
};
```

just like


```javascript
const classes = {
  root: `${PREFIX}-root`
};
```

appears in `SpeciesCorrectEdit.js` `SpeciesCorrectErrorResults.js` `SpeciesCorrectResults.js` and `SurveyDiff.js`

```javascript
const classes = {
  root: `${PREFIX}-root`
};
```

```javascript
const classes = {
  root: `${PREFIX}-root`
};
```

```javascript
const classes = {
  root: `${PREFIX}-root`
};
```

```javascript
const classes = {
  root: `${PREFIX}-root`
};
```

----

above are duplicate codes from your PR, which you can clearly able to identify a syntax

```javascript
const classes = {
  <element>: <unique-name>,
  ...
};
```

Too many duplicate codes to be acceptable, maybe a helper function for generate unique class names for different components if the class name generator from MUI not fit? https://mui.com/material-ui/experimental-api/classname-generator/

if you do go with a helper function let's say something like this should reduce a lot of duplicate code @HavierD 

```javascript
// Define a helper function to generate unique class names
function generateUniqueClasses(prefix, classNames) {
  const generatedClasses = {};
  
  for (const className of classNames) {
    generatedClasses[className] = `${prefix}-${className}`;
  }

  return generatedClasses;
}

// Usage example
const PREFIX = "yourPrefix";

const aClasses = generateUniqueClasses(PREFIX, ["fishSize", "invertSize"]);
// OR
const bClasses = generateUniqueClasses(PREFIX, ["root"]);

console.log(aClasses);
console.log(bClasses);
```

Reply:
Your solution has such side effects:

- Helper is good for extracting a complicated block, and use them with much simple ways. But in this case, it only change this: 
- 
```javascript

const classes = {root: `${PREFIX}-root`};

```

      into this:

```javascript

const bClasses = generateUniqueClasses(PREFIX, ["root"]);

```

        and also with an extra helper class.


- introducing an external library only for several string constants. NRMN now doesn't use uuid. Introducing it directly is not a good idea if it can only be replaced by several string declaration. 
 https://medium.com/@hello.shahanshah/problems-of-using-too-many-libraries-and-how-to-avoid-them-dccf9ac8e029
 https://www.linkedin.com/pulse/best-practices-using-javascript-libraries-frameworks-dibor-alfred/

-  uuid generates meaningless long random characters such as 104u1340qijfeaor324nj, but declaring PREFIX directly is meaningful. For example, if someone want to debug using DevTools Elements, the corresponding css class is like this: 
![image](https://github.com/aodn/nrmn-application/assets/102270719/d02f8237-ab1f-46bc-ac54-647e13ecea2d)
which is much readable and maintainable than random characters like: 104u1340qijfeaor324nj-root. No one can easily know which component this meaningless-character CSS class is in relation to. 

- Split the same portion of code. Currently, the code is like this: 
```javascript
const PREFIX = 'GridHeader';

const classes = {
  fishSize: `${PREFIX}-fishSize`,
  invertSize: `${PREFIX}-invertSize`
};

const Root = styled('div')(({theme}) => ({
  [`& .${classes.fishSize}`]: {
    color: '#c4d79b',
    borderBottom: '1px solid ' + theme.palette.divider
  },

  [`& .${classes.invertSize}`]: {
    color: '#da9694'
  }
}));

```
Which is very readable and no comments needed. Developers are easy to know: first variable is taking the component name as the prefix, second object is defining the css class names using the right above prefix, and the third object is defining the css values using the classes right above as a part of selectors. 

Thank you for your concern, but I think, in all the comments, my solution is better than others (as i explained above). Besides, not all similar structure needs to be extracted, and not all constants need to be assembled into a constant file. Using meaningless class name to instead meaningful class name is not good code practice; exposing a private member (which is dangerous to be public) is even worse code practice, etc. I accept all the recommendations which is better than mine (as I did in a lot of tickets), but currently i don't think this ticket needs to change, otherwise @utas-raymondng wouldn't approve my PR. I don't think my code of this ticket is bad and worth so much time to explain, I won't create a new pull request. If you still insist on your point and think my explanation is useless, please ask @vfisaac and @utas-raymondng for help to review all the points of us in this page, and make final decision.

Reply:
You have 2 samples, one with uuid and one not. Whatever ways, they are my suggestions to improve your way of duplicating code, I'm not fixing the PR for you with actual implementation, you need to workout and brainstorm a final way to improve it, but not duplicating code like you are using.

I will not satisfied with a bunch of duplicated lines with the reasons you mentioned.

I leave this up to the bosses @utas-raymondng @vfisaac

Reply:
They are not the common duplicating code as you reckoned. They are just one string declaration and one object declaration, and they must be private and these two are a part of the CSS block for a component. Private members with similar structure almost occurs everywhere in any projects. So I don't think I need to workout anyway to improve it.

Reply:
> They are not the common duplicating code as you reckoned. They are just one string declaration and one object declaration, and they must be private and these two are a part of the CSS block for a component. Private members with similar structure almost occurs everywhere in any projects. So I don't think I need to workout anyway to improve it.

The helper function itself can be public while the data it generates or processes remains private within the component where it's used. Your code are unnecessarily duplicated.

Reply:
Hi @HavierD, As an outcome of the meeting with other dev, please link this PR to either your new improvement PR or backlog ticket that you will work on to refactor the code, so the issue here is traceable for future reference. Thank you.

Reply:
Thank you for the reminding. I will do it shortly.